### PR TITLE
Add phpdoc for throws - Link

### PR DESCRIPTION
**Pull request by Matt75**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add phpdoc throws tag to help developer to identify potential error cases
| Type?         | refacto
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShopCorp/ps_mbo/pull/96
| How to test?  | Everything should works as before

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19450)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: classes/Link.php [hunk 1]
```diff
@@ -81,6 +86,8 @@ public function __construct($protocolLink = null, $protocolContent = null)
      * @param int $idPicture ID of the picture to delete
      *
      * @return string
+     *
+     * @throws PrestaShopException
```
Comment (PierreRambaud) [inline classes/Link.php]: Why adding this, this method doesn't throw PrestaShopException :thinking:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I'm against adding `@throws` when the method itself is not throwing exceptions, otherwise, we must change ALL methods to say it throws `Exception`, `DbExpection`, `PDOException`, etc. Which is illogic :/

File: classes/Link.php (hunk 1 above)
Comment (Matt75) [reply classes/Link.php]: PHPStorm inspection ;)

Comment (Matt75) [reply classes/Link.php]: Due to usage of Configuration::get() can return PrestaShopException on call Db::executeS

Comment (Matt75) [reply classes/Link.php]: I will update Pull Request according your feedback

Comment (PierreRambaud) [reply classes/Link.php]: Phpstorm :vomiting_face:

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Still against this kind of change, sorry, methods are not throwing exceptions.
Structural elements are not properly defined, many exceptions are missing and if we follow phpdoc
> This tag is used to present in your documentation which error COULD occur and under which circumstances. It is RECOMMENDED to provide a description that describes the reason an exception is thrown.

Otherwise, this annotation has no aim

**Comment by Matt75** [comment]
> Still against this kind of change, sorry, methods are not throwing exceptions.
> Structural elements are not properly defined, many exceptions are missing and if we follow phpdoc
> 
> > This tag is used to present in your documentation which error COULD occur and under which circumstances. It is RECOMMENDED to provide a description that describes the reason an exception is thrown.
> 
> Otherwise, this annotation has no aim

Pull request updated to only use @throws when the method itself throws an exception
Add warning on phpdoc for other cases about Symfony Router

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @Matt75

**Pull request merged**